File: dev/rabbitmq_ruby/send.rb
#!/usr/bin/env ruby
# encoding: utf-8

require "bunny"

conn = Bunny.new(:host=>"172.17.0.5", :port=>5672)
conn.start

ch = conn.create_channel
x = ch.topic("david")
routing_key = "hello"
routing_key2 = "world"
msg = "Hello World!"
msg2 = "World Hello!"

x.publish(msg, :routing_key => routing_key)
puts " [x] Sent '#{msg}'"
x.publish(msg2, :routing_key => routing_key2)
puts " [x] Sent '#{msg2}'"

conn.close

File: dev/rabbitmq_nodejs/send.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');

amqp.connect('amqp://172.17.0.5', function(err, conn) {
  conn.createChannel(function(err, ch) {
    var ex = 'david';
    var args = process.argv.slice(2);
    var key = 'hello';
    var msg = 'Hello World!';

    ch.assertExchange(ex, 'topic', {durable: false});
    ch.publish(ex, key, new Buffer(msg));
    console.log(" [x] Sent %s:'%s'", key, msg);
  });

  setTimeout(function() { conn.close(); process.exit(0) }, 500);
});
